### PR TITLE
[LLM][NPU] Ported sampler from Stateless to Stateful pipeline

### PR DESCRIPTION
**Pull request by AsyaPronina**
- *Ported sampler functionality from Stateless to Stateful pipeline*

### CODE REVIEW

File: src/cpp/src/llm_pipeline_static.cpp [hunk 1]
```diff
@@ -856,26 +864,41 @@ EncodedResults StatefulLLMPipeline::generate(
...
 
-    int64_t last_token = utils::argmax(m_request.get_tensor("logits"), 0);
+    auto logits = m_request.get_tensor("logits");
+    int64_t output_sequence_len = logits.get_shape().at(1);
 
-    results.tokens[0].push_back(last_token);
-    if (streamer_ptr && streamer_ptr->put(last_token)) {
-        return results;
-    }
+    // Swap max_new_token to get_max_new_token()
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline_static.cpp:870]: is this comment is valid? here we don't play with max_new_tokens

File: src/cpp/src/llm_pipeline_static.cpp [hunk 2]
```diff
@@ -856,26 +864,41 @@ EncodedResults StatefulLLMPipeline::generate(
...
+    auto sequence_group = std::make_shared<SequenceGroup>(
+        0 /* request_id */, input_ids, config, 1 /* block_size */);
+    sequence_group->update_processed_tokens_num(input_ids.get_size());
```
Comment (ilya-lavrenov) [inline src/cpp/src/llm_pipeline_static.cpp:873]:
```suggestion
    sequence_group->update_processed_tokens_num(sequence_group->get_prompt_len() - output_sequence_len);
```

Comment (TolyaTalamanov) [reply src/cpp/src/llm_pipeline_static.cpp:873]: Will it work w/o SLICE_OUT?

**Review by TolyaTalamanov — COMMENTED** [review]
LGTM, could you also enable testing of `StatefulLLMPipeline` in `test_llm_pipeline_static.py`?

File: src/cpp/src/llm_pipeline_static.cpp (hunk 2 above)
Comment (ilya-lavrenov) [reply src/cpp/src/llm_pipeline_static.cpp:873]: I suppose yes.

w/o Slice output len is the same as prompt len and, hence, number of processed tokens is 0

Comment (AsyaPronina) [reply src/cpp/src/llm_pipeline_static.cpp:873]: I agree, however, for first input_ids we have length == length of prompt, for example, `18`. But `output_sequence_len` is equal to 1024 for first `logits` as it is output from prefill model

**Review by ilya-lavrenov — APPROVED** [review]
